### PR TITLE
Fixes to TypeScript type definition file

### LINKED ISSUE #296: Typescript file issues
It seems like the typings file should be named index.d.ts and then you don't need to include a types: node in the package.json.  Otherwise a types node is needed.  Also, I believe a handful of exports are needed in the .t.ds definition files.  Currently the module isn't exporting anything.

### PR DESCRIPTION
**Pull request by JeffJacobson**
Fixes #296

### CODE REVIEW

File: test.ts [hunk 1]
```diff
@@ -1,10 +1,10 @@
-import Terraformer = require("./terraformer");
+import * as Terraformer from "./index";
 
 console.assert(typeof Terraformer !== undefined);
 
 let point = new Terraformer.Primitive({
   type: "Point",
-  coordinates: [1, 2]
+  coordinates: [1, 2],
```
Comment (jgravois) [inline test.ts]: are the new trailing commas added automatically?

Comment (jgravois) [inline test.ts]: once we clean them up, i think we'll be good to merge.

Comment (JeffJacobson) [inline test.ts]: I've added a (TSLint)[http://palantir.github.io/tslint/] test to `npm run test:ts`. The tslint config has been modified to disallow the trailing commas. (By default it requires them. I have no preference either way on this matter.)

Comment (jgravois) [inline test.ts]: you're a hero. i'll find time to pull this down and kick the tires in the next couple of days.

**Pull request merged**